analyzer/analyzer.go

```diff
@@ -71,8 +71,8 @@ func run(pass *analysis.Pass) (interface{}, error) {
 				}
 
 				if allowErrorInDefer {
-					if ident, ok := p.Type.(*ast.Ident); ok {
-						if ident.Name == "error" && findDeferWithErrorAssignment(funcBody, n.Name) {
+					if pass.TypesInfo.TypeOf(p.Type).String() == "error" { // with package prefix, so only built-it error fits
```

Comment:
nice catch! Could we use smth. like 
`if pass.TypesInfo.TypeOf(p.Type) == reflect.TypeOf(err)` or smth like this to not rely on string comparisons of the type?

Reply:
Don't know if it's possible to avoid strings at all, but found following example
https://github.com/golang/tools/blob/93a03c2c548821d80e09ec6159b80b6c8bee8887/go/analysis/passes/errorsas/errorsas.go#L81

Also they say to use `types.Identical` instead of `==` ([link](https://github.com/golang/example/tree/master/gotypes#types)):

> With the exception of `Named` types, instances of `Type` are not canonical. That is, it is usually a mistake to compare types using `t1==t2` since this equivalence is not the same as the [type identity relation](https://golang.org/ref/spec#Type_identity) defined by the Go spec.

So I've changed PR:

```go
errorType := types.Universe.Lookup("error").Type()
...
	if types.Identical(pass.TypesInfo.TypeOf(p.Type), errorType) {
```

Reply:
awesome thanks again!